addon/-private/system/model/internal-model.js

```diff
@@ -16,11 +16,22 @@ import {
   HasManyReference
 } from "ember-data/-private/system/references";
 
-var Promise = Ember.RSVP.Promise;
-var get = Ember.get;
-var set = Ember.set;
-var copy = Ember.copy;
-var assign = Ember.assign || Ember.merge;
+const {
+  get,
+  set,
+  copy,
+  Error: EmberError,
+  inspect,
+  isEmpty,
+  isEqual,
+  run: emberRun,
+  RSVP,
+  RSVP: { Promise }
+} = Ember;
+
+const assign = Ember.assign || Ember.merge;
+const setOwner = Ember.setOwner;
+const TransitionChainMap = new EmptyObject();
```

Comment:
Can you put a comment here or in the method as to how the transitionchain map works